### PR TITLE
validate driver via YAML files

### PR DESCRIPTION
**Pull request by pohly**
This fixes some issues in the objects (both those created via YAML files and the operator), makes them more similar to allow easier comparison, and finally compares against those YAML files when testing the operator instead of using hand-written, incomplete code.

### CODE REVIEW

File: pkg/deployments/load.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+			if container["name"].(string) == "pmem-ns-init" {
+				cmd := container["command"].([]interface{})
+				cmd = append(cmd, fmt.Sprintf("--useforfsdax=%d", deployment.Spec.PMEMPercentage))
+				container["command"] = cmd
+				break
+			}
+		}
+	}
+	for _, container := range containers {
+		// Mimick the current operator behavior
```
[the remark this answers is not in the document]
Comment (pohly) [reply pkg/deployments/load.go:137]: This is valid spelling - https://en.wiktionary.org/wiki/mimick

File: pkg/deployments/load.go [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+		*yaml = bytes.ReplaceAll(*yaml,
+			[]byte("imagePullPolicy: Always"),
+			[]byte("imagePullPolicy: "+deployment.Spec.PullPolicy))
+
+		*yaml = bytes.ReplaceAll(*yaml,
+			[]byte("-v=3"),
+			[]byte(fmt.Sprintf("-v=%d", deployment.Spec.LogLevel)))
+
+		*yaml = pmemImage.ReplaceAll(*yaml, []byte("image: "+deployment.Spec.Image))
```
[the remark this answers is not in the document]
Comment (pohly) [reply pkg/deployments/load.go:56]: It replaces the old PMEM-CSI image name with the intended one from the spec. `pmemImage` is a regular expression.

File: pkg/deployments/load_test.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	"github.com/intel/pmem-csi/pkg/deployments"
+	"github.com/intel/pmem-csi/pkg/version"
+)
+
+func TestLoadObjects(t *testing.T) {
+	_, err := deployments.LoadObjects(version.NewVersion(1, 0), api.DeviceModeDirect)
+	assert.Error(t, err, "load yaml for unsupported version")
+
+	yamls := deploy.ListAll()
+	assert.NotEmpty(t, yamls, "should have builtin yaml depoyments")
```
[the remark this answers is not in the document]
Comment (pohly) [reply pkg/deployments/load_test.go]: Ack. I'll fix it before merging.

**Comment by pohly** [comment]
@avalluri: the updated commits also passed testing. Okay to merge?

**Comment by avalluri** [comment]
> @avalluri: the updated commits also passed testing. Okay to merge?

Yes.

**Pull request merged**